### PR TITLE
Add function to apply conversion rules to generated markdown links

### LINKED ISSUE #54: Need to apply text-to-link conversion rules in Markdown generation
Github applies a set of text conversion rules to markdown header text when converting it into a link.

The markdown generation script converts standard name dictionary sections into markdown headers and header links, but doesn't apply these conversion rules, which can result in broken links.   Thus the `write_standard_name_table.py` script should be modified to apply those rules when writing the markdown file.

### PR DESCRIPTION
**Pull request by nusbaume**
This PR adds a function to apply header text conversion rules when generating internal markdown document links.

Fixes #54

### CODE REVIEW

**Comment by nusbaume** [comment]
Please note that these test failures should be fixed once PR #52 has been accepted and merged (both into main and my branch).

File: tools/write_standard_name_table.py [hunk 1]
```diff
@@ -25,10 +25,40 @@
...
+    text conversion rules.  This function thus
+    applies those same rules to a given string
+    in order to produce the correct link.
+    """
+
+    #First trim the string to remove leading/trailing white space:
+    link_str = text_str.strip()
+
+    #Next, make sure all text is lowercase:
+    link_str = link_str.lower()
```
Comment (mkavulich) [inline tools/write_standard_name_table.py:51]: Is this check necessary? It appears as if the old links with capital letters still work in the web view, but maybe there's another markdown reader where it doesn't work?

**Review by mkavulich — APPROVED** [review]
Looks good! Just had one question.

File: tools/write_standard_name_table.py (hunk 1 above)
Comment (nusbaume) [reply tools/write_standard_name_table.py:51]: That's a good question!  After doing some experimentation it looks like at least my web browser (Chrome) is able to use a URL with uppercase text correctly, but I think "under the hood" the actual link is still all lowercase.  You can see this if you go to to one of the section heads in the markdown document itself and then click on the the little chain symbol that pops up next to it.

So, I am happy to remove this text conversion if that is preferred by you or others, but I personally think keeping everything lowercase will be the safest option in the long run.  Of course please let me know if you disagree!

**Comment by nusbaume** [comment]
@mkavulich I removed the "needs discussion" label on this PR because I _think_ you meant to add it to PR #56 instead based off of my memory of Monday's framework meeting discussions.  However, if I am wrong here then feel free to add the label back.  Thanks!

**Comment by mkavulich** [comment]
@nusbaume Yes, you're right I applied the label to the wrong PR. Thanks for fixing that!

File: tools/write_standard_name_table.py (hunk 1 above)
Comment (mkavulich) [reply tools/write_standard_name_table.py:51]: It's fine to leave the check in, I was just curious if it was causing issues.

**Review by grantfirl — APPROVED** [review]

**Pull request merged**